### PR TITLE
feat(number-field): add decimalSeparator prop #596

### LINKED ISSUE #596: [NumberField]: NumberField fails to handle localized decimal separators on mobile browsers with Estonian locale
## Problem
NumberField does not handle localized decimal separators correctly on mobile. On iOS with Estonian locale, the decimal keyboard provides , instead of ., and the current native type="number" input does not persist that value as expected. This makes decimal entry unreliable on iOS browsers and potentially inconsistent on Android browsers as well.

## Solution
Update NumberField so decimal input is parsed in component logic instead of relying on native type="number" behavior.
Implemented approach:
• Replace native number parsing with controlled text input handling
• Accept both . and , as decimal separators
• Normalize , to .
• Preserve intermediate typing states such as 1, and 1.
• Allow consumers to explicitly use inputMode="decimal" when they want a decimal keyboard on mobile
This keeps existing API behavior stable while fixing localized decimal entry where the keyboard provides a comma

## DoD
- [ ] The component is developed using best practices, conventions, and modern frontend trends
- [ ] The component follows **BEM naming conventions** (if applicable)
- [ ] The component implementation **matches the Figma design**, including all defined states and interactions
- [ ] The component supports **responsiveness** and scales across breakpoints (if applicable)
- [ ] The component does not introduce **runtime errors or warnings**
- [ ] Linter passes
- [ ] Component has export if needed
- [ ] The component has **unit tests** with at least **80% code coverage**
- [ ] CodeCov passes minimum coverage threshold
- [ ] The component has passed **code review**
- [ ] The component has passed **Chromatic visual review**
- [ ] **Storybook stories** are created:
  - [ ] Cover all Figma variations
  - [ ] Include edge cases, empty/error states
- [ ] Component’s **API (props, slots, events)** is documented
- [ ] **ZeroHeight is updated** with design/development guidelines
- [ ] **External dependencies** used in the component are documented in Confluence with rationale.
- [ ] **Semantic commit format** is used (for `semantic-release`)
- [ ] External libraries **comply with [validated open source licenses](https://github.com/TEHIK-EE/tedi-design-system/wiki/Definition-of-Done#validated-open-source-component-licences)**
- [ ] Component is versioned correctly and ready for release if required
- [ ] Update [statuses page in Zeroheight](https://www.tedi.ee/1ee8444b7/p/300e17-komponentide-staatused)

### PR DESCRIPTION
**Pull request by airikej**
## Summary by CodeRabbit

* **New Features**
  * Locale-aware number input with configurable decimal separator (comma or dot), improved editing/formatting, and refined decimal handling.
  * Clearing the field now results in an empty value state.
  * Label context now exposes the active locale (defaults to English) for consistent formatting.

* **Accessibility**
  * Improved spinbutton semantics and aria attributes for number fields.

* **Tests**
  * Expanded coverage for parsing, formatting, edge cases, partial input, and clear/blur behaviors.

[Review Change Stack](https://app.coderabbit.ai/change-stack/TEDI-Design-System/react/pull/623?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/tedi/components/form/number-field/number-field.tsx [hunk 1]
```diff
@@ -85,8 +92,9 @@ export const NumberField = (props: NumberFieldProps) => {
     required,
     className,
     size,
-    inputMode = 'numeric',
+    inputMode,
     decimalPlaces,
+    decimalSeparator = '.',
```
Comment (mart-sessman) [inline src/tedi/components/form/number-field/number-field.tsx]: kas meil mingit providerit ei ole, kus saaks seda seadistada? Praegu peaks iga komponendi juures ütlema decimalSeparator=","

File: src/tedi/components/form/number-field/number-field.tsx [hunk 2]
```diff
@@ -157,11 +183,30 @@ export const NumberField = (props: NumberFieldProps) => {
...
     onChange?.(returnValue);
     setInputInnerValue(returnValue);
+    setDisplayValue(formatNumber(returnValue));
+  };
+
+  const handleInputChange = ({ currentTarget: { value: rawValue } }: ChangeEvent<HTMLInputElement>) => {
+    setDisplayValue(rawValue);
+
+    const normalized = rawValue.replace(',', '.');
+    const parsed = rawValue === '' ? 0 : parseFloat(normalized);
```
Comment (mart-sessman) [inline src/tedi/components/form/number-field/number-field.tsx]: kas on hea mõte defaultida 0 peale tühja väärtuse korral? või peaks emittima undefined/null/NaN?

File: src/tedi/components/form/number-field/number-field.tsx [hunk 3]
```diff
@@ -157,11 +183,30 @@ export const NumberField = (props: NumberFieldProps) => {
     updateValueUpdatedLabel(returnValue);
     onChange?.(returnValue);
     setInputInnerValue(returnValue);
+    setDisplayValue(formatNumber(returnValue));
+  };
+
+  const handleInputChange = ({ currentTarget: { value: rawValue } }: ChangeEvent<HTMLInputElement>) => {
```
Comment (mart-sessman) [inline src/tedi/components/form/number-field/number-field.tsx:196]: vb ei peaks onChange trigerdama, kui kasutaja trükkis separator char'i, et me ei tulistaks sama väärtust topelt. Praegu nt user trükib
"2" -> onChange(2)
"2," -> onChange(2)
2,5 -> onChange(2.5)

File: src/tedi/components/form/number-field/number-field.tsx [hunk 4]
```diff
@@ -101,18 +109,36 @@ export const NumberField = (props: NumberFieldProps) => {
...
   const inputRef = useRef<HTMLInputElement>(null);
+  const isFocusedRef = useRef(false);
 
   const [inputUpdated, setInputUpdated] = useState<string>('');
   const [inputInnerValue, setInputInnerValue] = useState<number>(defaultValue ?? 0);
+  const [displayValue, setDisplayValue] = useState<string>(() =>
+    formatNumber(value !== undefined ? value : defaultValue ?? 0)
+  );
 
   const getCurrentValue = useMemo(
```
Comment (mart-sessman) [inline src/tedi/components/form/number-field/number-field.tsx]: miks useMemo? useCallback oleks fn puhul parem

**Review by mart-sessman — CHANGES_REQUESTED** [review]

File: src/tedi/components/form/number-field/number-field.tsx (hunk 4 above)
Comment (airikej) [reply src/tedi/components/form/number-field/number-field.tsx]: ai ütles et useCallback oleks vale kasutada siin, kuid useMemo on ka overkill, et tava const parem

File: src/tedi/components/form/number-field/number-field.tsx (hunk 3 above)
Comment (airikej) [reply src/tedi/components/form/number-field/number-field.tsx:196]: fixed

**Review by mart-sessman — APPROVED** [review]
